ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Offset.scala

```diff
@@ -36,6 +36,8 @@ object Offset {
 
   val beforeBegin: Offset = Offset.fromByteArray(Array.empty[Byte])
 
+  val begin: Offset = Offset.fromByteArray(Array(0: Byte))
```

Comment:
I seem to remember that some ledgers might not use `0` as the beginning. I'm not sure if this is an issue; it might be OK because it's the lowest possible offset, even if there isn't a `0`.

@simonmaxen-da, can you comment?

Reply:
This is basically what we used to return before this:

https://github.com/digital-asset/daml/commit/12c05c408af5abefad56d1135aa9d36ee1c2ea3c

So I hope it's not problematic from a functional behavior perspactive, at least the commit message in that commit doesn't suggest that it was

Reply:
@SamirTalwar-DA - the first used offset should be at the discretion of the ledger implementor.  With the current DAML-on-Corda offset calculation the first offset will not be Offset.begin.   In future it could be.

Reply:
The ledger API server doesn't expose the underlying ledger's offsets; see the `parameters` table of the indexer and the `ledger_end` and `external_ledger_end` column. So IIUC, whatever the underlying ledger uses as the first offset, `0` should be consistent with what the Ledger API server uses.

Reply:
That makes sense. This line probably belongs in `ApiOffset`, not here, then.

Reply:
Moved to `ApiOffset` then